compiler/src/dmd/common/charactertables.d

```diff
@@ -0,0 +1,267 @@
+/**
+ * Character tables related to identifiers.
+ *
+ * Supports UAX31, C99, C11 and least restrictive (All).
+ *
+ * Copyright: Copyright (C) 1999-2024 by The D Language Foundation, All Rights Reserved
+ * Authors:   $(LINK2 https://cattermole.co.nz, Richard (Rikki) Andrew Cattermole)
+ * License:   $(LINK2 https://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Source:    $(LINK2 https://github.com/dlang/dmd/blob/master/src/dmd/common/charactertables.d, common/charactertables.d)
+ * Documentation: https://dlang.org/phobos/dmd_common_charactertables.html
+ * Coverage:    https://codecov.io/gh/dlang/dmd/src/master/src/dmd/common/charactertables.d
+ */
+module dmd.common.charactertables;
+
+@safe nothrow @nogc pure:
+
+extern(C++):
+
+///
+enum IdentifierTable {
+    UAX31, ///
+    C99, ///
+    C11, ///
+    LR, /// Least Restrictive aka All
+}
+
+///
+struct IdentifierCharLookup
+{
+    @safe nothrow @nogc pure:
+
+    ///
+    extern(C++) bool function(dchar) isStart;
+    ///
+    extern(C++) bool function(dchar) isContinue;
+
+    /// Lookup the table given the table name
+    static IdentifierCharLookup forTable(IdentifierTable table)
+    {
+        import dmd.common.identifiertables;
+
+        // Awful solution to require these lambdas.
+        // However without them the extern(C++) ABI issues crop up for isInRange,
+        //  and then it can't access the tables.
+        final switch(table) {
+            case IdentifierTable.UAX31:
+                return IdentifierCharLookup(
+                        (c) => isInRange!UAX31_Start(c),
+                        (c) => isInRange!UAX31_Continue(c));
+            case IdentifierTable.C99:
+                return IdentifierCharLookup(
+                        (c) => isInRange!FixedTable_C99_Start(c),
+                        (c) => isInRange!FixedTable_C99_Continue(c));
+            case IdentifierTable.C11:
+                return IdentifierCharLookup(
+                        (c) => isInRange!FixedTable_C11_Start(c),
+                        (c) => isInRange!FixedTable_C11_Continue(c));
+            case IdentifierTable.LR:
+                return IdentifierCharLookup(
+                        (c) => isInRange!LeastRestrictive_Start(c),
+                        (c) => isInRange!LeastRestrictive_Continue(c));
+        }
+    }
+}
+
+/**
+Convenience function for use in places where we just don't care,
+what the identifier ranges are, or if it is start/continue.
+
+Returns: is character a member of least restrictive of all.
+*/
+bool isAnyIdentifierCharacter(dchar c)
+{
+    import dmd.common.identifiertables;
+    return isInRange!LeastRestrictive_OfAll(c);
+}
+
+///
+unittest
+{
+    assert(isAnyContinue('ğ'));
```

Comment:
I think you wanted to test `isAnyIdentifierCharacter` here.
`isAnyContinue` is defined and tested later

Reply:
Indeed.